server/lib/notifications.js

```diff
@@ -137,6 +141,61 @@ export async function notifyAdminsOfCollective(CollectiveId, activity, options =
   return notifySubscribers(adminUsers, activity, options);
 }
 
+async function notifyAdminsOfCollectiveAndCheckUserExpensesAmount(activity) {
+  try {
+    await notifyAdminsOfCollective(activity.data.collective.id, activity);
+    await notifyUserIfExpensesExceededMinimumFormAmount(activity);  
+  } catch (error) {
+    console.log(error);
+  }
+  
+}
+      
+async function notifyUserIfExpensesExceededMinimumFormAmount(activity) {
+  try {
+    const host = await models.Collective.findById(activity.data.host.id);
+    if (host.currency && host.currency !== 'USD') {
+      return;
+    }
+    // ideally change to consider the current fiscal year instead(Example: might be from
+    // march to february instead of from the beginning from january)
+    const beginningOfCurrentYear = new Date(new Date().getFullYear(), 0, 1);
+    const userExpenses = await models.Expense.findAll({
+      where: {
+        UserId: activity.data.user.id,
+        createdAt: { [Op.gte]: beginningOfCurrentYear },
+      }
+    });
+    let totalAmount = 0;
+    for (const expense of userExpenses) {
+      let expenseAmountInHostCurrency = expense.amount;
+      if (expense.currency !== host.currency) {
+        const expenseDate = (expense.status === expenseStatuses.PAID) ? expense.updatedAt : 'latest';
+        expenseAmountInHostCurrency = await convertToCurrency(expense.amount, expense.currency, host.currency, expenseDate);
+      }
+      totalAmount += expenseAmountInHostCurrency;
+    }
+    // U$ 600.00 total amount allowed without form as of July 2018
+    if (totalAmount > MAXIMUM_AMOUNT_TO_SEND_FORM_COMMENT) {
+      const fromCollectiveIdData = await models.Collective.findOne({ 
+        where: { 
+          name: OPENCOLLECTIVE_BOT_NAME
+        }
+      });
+      const commentData = {
+        CollectiveId: activity.data.collective.id,
+        ExpenseId: activity.data.expense.id,
+        html: expensesFormData.FORM_EXPENSE_COMMENT_HTML,
+        FromCollectiveId: (fromCollectiveIdData ? fromCollectiveIdData.id : null)
+      };
+      return models.Comment.create(commentData);
+    }
+    return Promise.resolve(true);
+  } catch (error) {
+    throw error;
```

Comment:
In a `try...catch` block, the error is caught if it is thrown in the `try` block so throwing it again after catching it is redundant. Is there a need for this?

Reply:
You're right, as said in my comment I believe we should either just log it or log and warn devs if it feels like it's important, but at the very least log it. Will fix.